pkg/kubernetes/persistentvolumeclaim/v1alpha1/persistentvolumeclaim.go

```diff
@@ -180,3 +183,94 @@ func (b *ListBuilder) WithFilter(pred ...Predicate) *ListBuilder {
 	b.filters = append(b.filters, pred...)
 	return b
 }
+
+// NewPVC returns a PVC instance
+func NewPVC() *PVC {
+	return &PVC{Object: &corev1.PersistentVolumeClaim{}}
+}
+
+// WithName returns a PVC instance with name
+func (p *PVC) WithName(name string) *PVC {
+	if p.Err != nil {
+		return p
+	}
+	if len(name) == 0 {
+		p.Err = errors.New("PVC name shouldn't be empty")
+		return p
+	}
+	p.Object.Name = name
+	return p
+}
+
+// WithNamespace returns a PVC instance with namespace
+func (p *PVC) WithNamespace(namespace string) *PVC {
+	if p.Err != nil {
+		return p
+	}
+	if len(namespace) == 0 {
+		namespace = "default"
+	}
+	p.Object.Namespace = namespace
+	return p
+}
+
+// WithAnnotations returns a PVC instance with annotations
+func (p *PVC) WithAnnotations(annotations map[string]string) *PVC {
+	if p.Err != nil {
+		return p
+	}
+	p.Object.Annotations = annotations
+	return p
+}
+
+// WithLabels returns a PVC instance with labels
+func (p *PVC) WithLabels(labels map[string]string) *PVC {
+	if p.Err != nil {
+		return p
+	}
+	p.Object.Labels = labels
+	return p
+}
+
+// WithStorageClass returns a PVC instance with storageclass
+func (p *PVC) WithStorageClass(scName string) *PVC {
+	if p.Err != nil {
+		return p
+	}
+	if len(scName) == 0 {
+		p.Err = errors.New("PVC storageclass name shouldn't be empty")
+		return p
+	}
+	p.Object.Spec.StorageClassName = &scName
+	return p
+}
+
+// WithAccessModes returns a PVC instance with annotations
+func (p *PVC) WithAccessModes(accessMode []corev1.PersistentVolumeAccessMode) *PVC {
+	if p.Err != nil {
+		return p
+	}
+	if len(accessMode) == 0 {
+		p.Err = errors.New("PVC accessMode shouldn't be empty")
+		return p
+	}
+	p.Object.Spec.AccessModes = accessMode
+	return p
+}
+
+// WithCapacity returns a PVC instance with capacity
+func (p *PVC) WithCapacity(capacity string) *PVC {
```

Comment:
This is not what idiomatic maya follows?
Why are we trying to deviate from our standard?

Reply:
Refactored the code